File: backend/app/schemas/document.py
# These schemas define the input and output shapes for documents.

from pydantic import BaseModel


class DocumentResponse(BaseModel):
    id: int
    title: str
    file_name: str
    storage_path: str
    document_type: str
    status: str
    uploaded_by_user_id: int | None

    model_config = {"from_attributes": True}

File: frontend/src/types/document.ts
export interface DocumentItem {
  id: number;
  title: string;
  file_name: string;
  storage_path: string;
  document_type: string;
  status: string;
  uploaded_by_user_id: number | null;
}

export interface ApiSuccessResponse<T> {
  success: true;
  message: string;
  data: T;
}